12-http/example3/main.go

```diff
@@ -18,20 +21,46 @@ type user struct {
 	Phone string
 }
 
-// users is a slice of users.
-var users []user
+var (
+	// mu protects users.
+	mu sync.RWMutex
+	// initialize users, so that the output JSON won't be null
+	// prior to adding users.
+	users = make([]user, 0)
+)
+
+// index page template
+var IdxTpl = template.Must(template.ParseFiles("template/index.html"))
+
```

Comment:
Nope. It's package main though, so except in rare circumstances, exporting rules are more gray. I'll unexport it though.